src/control/common/proc_utils.go

```diff
@@ -0,0 +1,89 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package common
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func getProcPids(procDir, procName string) (pids []int, _ error) {
+	allProcs, err := filepath.Glob(procDir + "/*/status")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read process list")
+	}
+	for _, proc := range allProcs {
+		data, err := ioutil.ReadFile(proc)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to read process command line")
+		}
+		lines := strings.Split(string(data), "\n")
+		if len(lines) == 0 {
+			return nil, errors.Errorf("invalid process status file: %s", proc)
+		}
+		namePair := strings.SplitN(lines[0], ":", 2)
+		if len(namePair) < 2 {
+			return nil, errors.Errorf("invalid process status file: %s", proc)
+		}
+		if strings.TrimSpace(namePair[1]) != procName {
```

Comment:
If `procName` is greater than 15 characters the `/proc/<pid>/status` file is not adapted as the process name will be trauncated. Thus, the comparaison should be limited to the first 15 bytes of the name or it could be better to use the `/proc/<pid>/cmdline` file.

Reply:
Hmm. This is a good point. Also, we should probably use the `/proc/<pid>/stat` file for info and use `fmt.Ssccanf()` for efficiency. I did initially look at the `cmdline` file but that gets into other weird parsing issues. Maybe we just use it as a fallback for long `procName` values.

Reply:
I think I'm just going to compare the first 15 characters. I played around with some other tricky scenarios and I think using cmdline could open a can of worms, e.g.
```
$ ps ax | grep long_name
3063571 pts/7    S+     0:00 ssh localhost /home/macdonal/daos/test_long_name_very_long
3063583 ?        Ssl    0:00 /home/macdonal/daos/test_long_name_very_long
[macdonal@wolf-61 ~]$ cat /proc/3063571/cmdline
sshlocalhost/home/macdonal/daos/test_long_name_very_long
```
I don't really want to get into trying to parse all of the variations here...

Reply:
Not sure to fully understand your example.
For me the result of the cat is OK as the seperator in the cmdline file are '\0' characters.